### PR TITLE
Bug 1766287: Improve usability of hello-openshift imagestream

### PR DESCRIPTION
**Pull request by yselkowitz**
- Set local lookup policy on CVO manifests
- Update hello-openshift pullspec in CVO manifests

/assign @adambkaplan

### CODE REVIEW

File: manifests/08-openshift-imagestreams.yaml [hunk 1]
```diff
@@ -145,6 +159,8 @@ metadata:
     include.release.openshift.io/single-node-developer: "true"
     include.release.openshift.io/single-node-production-edge: "true"
 spec:
+  lookupPolicy:
+    local: true
```
Comment (adambkaplan) [inline manifests/08-openshift-imagestreams.yaml:163]: openshift/oauth-proxy resolves to a Docker Hub image. There is a risk that an end-user is pulling the (obsolete) oauth-proxy image from Docker Hub today, and with this switch will start pulling the image from the internal registry.

https://hub.docker.com/r/openshift/oauth-proxy

File: manifests/08-openshift-imagestreams.yaml [hunk 2]
```diff
@@ -164,10 +180,12 @@ metadata:
...
```
Comment (adambkaplan) [inline manifests/08-openshift-imagestreams.yaml:184]: Likewise, openshift/hello-openshift resolves to a Docker Hub image.

**Review by adambkaplan — COMMENTED** [review]
/hold

@yselkowitz over Slack you asked, "Why don't we set `lookupPolicy: local: true` on everything?", and at the time I didn't have a good reason not to.

Now that the PR is in front of me, I have found a reason why we haven't set this field. Our convention for imagestream tag pull specs can overlap with actual Docker Hub images. In the case of `oath-proxy` and `hello-openshift` we would replace the docker hub image with the one from the payload on upgrade. To me, this is a breaking change.

Unfortunately this means that @cyril-ui-developer is out of luck here. His next best option is to add the `alpha.image.policy.openshift.io/resolve-names: '*'` annotation to the Deployment's pod template spec.

See https://docs.openshift.com/container-platform/4.6/openshift_images/using-imagestreams-with-kube-resources.html#images-managing-images-enabling-imagestreams-kube_using-imagestreams-with-kube-resources

File: manifests/08-openshift-imagestreams.yaml (hunk 1 above)
Comment (yselkowitz) [reply manifests/08-openshift-imagestreams.yaml]: That probably would be a good thing, but let's leave that for further discussion in a separate PR.

File: manifests/08-openshift-imagestreams.yaml (hunk 2 above)
Comment (yselkowitz) [reply manifests/08-openshift-imagestreams.yaml:168]: That is actually the entire point of the (new in 4.7) hello-openshift imagestream, to easily replace uses of the unmaintained, single-arch, and now rate-limited image on docker.io.

**Comment by yselkowitz** [comment]
Repushed the PR limiting the changes to hello-openshift for now.

**Review by adambkaplan — APPROVED** [review]
/hold cancel

/approve

**Comment by adambkaplan** [comment]
@yselkowitz please squash your commits.

**Comment by yselkowitz** [comment]
Squashed as requested.

**Review by adambkaplan — APPROVED** [review]
/lgtm

Tested via cluster-bot and the dev console. Image replacement still works when the `alpha.image.policy.openshift.io/resolve-names: '*'` annotation is removed.

**Comment by smarterclayton** [comment]
I asked for this image to be removed from the product in https://github.com/openshift/enhancements/pull/438 - I really don't want to be adding stuff around this.  Ingress should be using their operator image and a subcommand of their operator binary, not depending on a generic example image.

**Comment by gabemontero** [comment]
> I asked for this image to be removed from the product in [openshift/enhancements#438](https://github.com/openshift/enhancements/pull/438) - I really don't want to be adding stuff around this. Ingress should be using their operator image and a subcommand of their operator binary, not depending on a generic example image.

OK @smarterclayton I've subscribed to that EP (I see @yselkowitz already responded to you there on current usage of this imagestream) and ack that I'm now officially aware of your wishes

My first thought is that I need to invesitigate / understand when the existing multi arch / console / serverless uses will be replaced so as not to break them.  Or do you want this imagestream removed now as a way to trigger their replacement?

**Pull request merged**